tests/test_graph_provider_join.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import absolute_import
+
+from copy import deepcopy
+import logging
+import pymongo
+import unittest
+
+import daisy
+
+logger = logging.getLogger(__name__)
+daisy.scheduler._NO_SPAWN_STATUS_THREAD = True
+
+
+class TestMongoGraphProviderJoin(unittest.TestCase):
+
+    def setUp(self):
+        db_host = 'localhost'
+        self.db_name = 'test_daisy_graph'
+        try:
+            self.client = pymongo.MongoClient(
+                db_host, serverSelectionTimeoutMS=0)
+        except pymongo.errors.ServerSelectionTimeoutError:
+            self.skipTest("No MongoDB server found")
+
+    def delete_db(self):
+        self.client.drop_database(self.db_name)
+
+    def mongo_provider_factory(self, mode):
+        return daisy.persistence.MongoDbGraphProvider(
+            self.db_name,
+            mode=mode,
+            position_attribute=['x', 'y'])
+
+    def test_join_collection(self):
```

Comment:
Can you provide a brief high level description and expected behavior for each test to help me understand what you're testing?